application-ownership-models/README.md

```diff
@@ -0,0 +1,116 @@
+- The non-business hours on-call is currently shared with other teams from the same area (hence "almost" full ownership), which is done for the sake of not overwhelming on-callers with frequent night shifts.
+- Ideally, if a team has enough people and we have state-of-the-art monitoring and alerting, meaning that there are next-to-none false positive alerts (for example caused by alerts from other team's components), the team can own 24h on-call for the application.
+
+### 1.2. Partial application lifecycle ownership
+
+Partial application ownership means that application development is done outside of Giant Swarm, therefore this is the case of upstream applications that we are deploying to our clusters.
+
+To name a few examples, we run, but do not develop, Prometheus, Grafana, Cluster API controllers, Kyverno, different CNI, CSI, CPI apps, then core Kubernetes components, such as API server, scheduler, controller manager, etcd, kubelet, etc.
+
+Partial application ownership of these components means that a Giant Swarm team owns:
+- Release engineering, fully or partially;
+- Operations, fully or partially, depending if the team has sole or shared ownership of the application, which is discussed in the next section.
+
+**Release engineering is owned fully** in case of forked projects, where Giant Swarm team builds the application from the forked repository, then tests, packages and finally delivers it.
+
+**Release engineering is owned partially** when Giant Swarm team does not build the app, but, for example, re-tagged container image is used. Here we have two cases:
+- Giant Swarm team is packaging the app as a Helm chart in the app repository, by building required Helm templates for deploying re-tagged container images and optionally using existing upstream Kubernetes manifests instead of writing Helm templates from scratch.
+- Giant Swarm is repackaging or importing existing upstream Helm charts into Giant Swarm application repositories.
+
+## 2. Sole or shared application ownership
+
+While different application ownership models based on application lifecycle stages are relatively clear and it's obvious which model is used for every application, for some apps is much less evident which team should own them, and if there can even be just one owning team, or multiple teams cannot be avoided due to shared nature of the app.
+
+Additionally, while application lifecycle stages split the app ownership between external (upstream) and internal (Giant Swarm) teams, sole or shared ownership is about how we handle app ownership internally within Giant Swarm.
+
+### 2.1. Sole application ownership
+
+The is the classical and usually the obvious case. A Giant Swarm team develops the application, in case of full application lifecycle ownership described in the previous section, or the team takes care of release engineering and operations for the app, in case of partial application lifecycle ownership described above.
```

Comment:
```suggestion
This is the classical and usually the obvious case. A Giant Swarm team develops the application, in case of full application lifecycle ownership described in the previous section, or the team takes care of release engineering and operations for the app, in case of partial application lifecycle ownership described above.
```